Build failure: Error while generating Javadoc

After cloning the repository I ran `mvn clean install` which gave the following build error:

```[INFO] --- maven-javadoc-plugin:2.10.4:jar (attach-javadocs) @ microprofile-fault-tolerance-api ---
[INFO] 
Loading source files for package org.eclipse.microprofile.faulttolerance...
Loading source files for package org.eclipse.microprofile.faulttolerance.exceptions...
Constructing Javadoc information...
Standard Doclet version 11.0.5
Building tree for all the packages and classes...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/Asynchronous.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/Bulkhead.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/CircuitBreaker.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/ExecutionContext.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/Fallback.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/Fallback.DEFAULT.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/FallbackHandler.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/Retry.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/Timeout.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/exceptions/BulkheadException.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/exceptions/CircuitBreakerOpenException.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/exceptions/FaultToleranceDefinitionException.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/exceptions/FaultToleranceException.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/exceptions/TimeoutException.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/package-summary.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/package-tree.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/exceptions/package-summary.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/exceptions/package-tree.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/constant-values.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/serialized-form.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/class-use/CircuitBreaker.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/class-use/Retry.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/class-use/Asynchronous.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/class-use/ExecutionContext.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/class-use/Timeout.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/class-use/FallbackHandler.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/class-use/Fallback.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/class-use/Fallback.DEFAULT.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/class-use/Bulkhead.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/exceptions/class-use/BulkheadException.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/exceptions/class-use/FaultToleranceDefinitionException.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/exceptions/class-use/TimeoutException.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/exceptions/class-use/FaultToleranceException.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/exceptions/class-use/CircuitBreakerOpenException.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/package-use.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/org/eclipse/microprofile/faulttolerance/exceptions/package-use.html...
Building index for all the packages and classes...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/overview-tree.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/index-all.html...
Building index for all classes...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/allclasses-index.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/allpackages-index.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/deprecated-list.html...
Building index for all classes...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/allclasses.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/allclasses.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/index.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/overview-summary.html...
Generating /Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs/help-doc.html...
1 error
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for MicroProfile Fault Tolerance 2.1-SNAPSHOT:
[INFO] 
[INFO] MicroProfile Fault Tolerance ....................... SUCCESS [  1.358 s]
[INFO] microProfile-fault-tolerance-api ................... FAILURE [  3.244 s]
[INFO] microprofile-fault-tolerance-tck ................... SKIPPED
[INFO] microprofile-fault-tolerance-spec .................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.733 s
[INFO] Finished at: 2019-11-28T15:35:39Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.10.4:jar (attach-javadocs) on project microprofile-fault-tolerance-api: MavenReportException: Error while generating Javadoc: 
[ERROR] Exit code: 1 - javadoc: error - The code being documented uses modules but the packages defined in http://docs.oracle.com/javase/8/docs/api/ are in the unnamed module.
[ERROR] 
[ERROR] Command line was: /Library/Java/JavaVirtualMachines/adoptopenjdk-11-openj9.jdk/Contents/Home/bin/javadoc @options @packages
[ERROR] 
[ERROR] Refer to the generated Javadoc files in '/Users/josephcass/Documents/libertyGit/microprofile-fault-tolerance/api/target/apidocs' dir.
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :microprofile-fault-tolerance-api
```

I worked around the issue by updating the maven-javadoc-plugin veresion in the pom.xml to version 3.1.1. I tried this based on the information from the following Javadoc version bug: https://bugs.openjdk.java.net/browse/JDK-8212233 

My Java version: 
```$ java -version
openjdk version "11.0.5" 2019-10-15
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.5+10)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.17.0, JRE 11 Mac OS X amd64-64-Bit Compressed References 20191016_371 (JIT enabled, AOT enabled)
OpenJ9   - 77c1cf708
OMR      - 20db4fbc
JCL      - 2a7af5674b based on jdk-11.0.5+10)
```

--- a/pom.xml
+++ b/pom.xml
@@ -31,7 +31,7 @@
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
 
-        <checkstyle.version>2.17</checkstyle.version>
+        <checkstyle.version>3.1.0</checkstyle.version>
         <!-- whether autorelease maven central staging repositories - default true to allow review and manually release repositories -->
         <autorelease>false</autorelease>
         <!-- keeping closed repos with failure - default is false because the errors are visible in the maven output, but true will leave the repo open for investigation in Sonatype Nexus -->
@@ -137,17 +137,17 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-jar-plugin</artifactId>
-                    <version>3.0.2</version>
+                    <version>3.2.0</version>
                 </plugin>
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-source-plugin</artifactId>
-                    <version>3.0.1</version>
+                    <version>3.2.0</version>
                 </plugin>
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-javadoc-plugin</artifactId>
-                    <version>2.10.4</version>
+                    <version>3.1.1</version>
                 </plugin>
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
@@ -184,7 +184,7 @@
             <plugin>
                 <groupId>org.apache.rat</groupId>
                 <artifactId>apache-rat-plugin</artifactId>
-                <version>0.12</version>
+                <version>0.13</version>
                 <executions>
                     <execution>
                         <id>rat-check</id>

